packages/bloc/lib/src/cubit.dart

```diff
@@ -0,0 +1,176 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'bloc.dart';
+import 'bloc_observer.dart';
+import 'change.dart';
+
+/// {@template cubit_unhandled_error_exception}
+/// Exception thrown when an unhandled error occurs within a [Cubit].
+///
+/// _Note: thrown in debug mode only_
+/// {@endtemplate}
+class CubitUnhandledErrorException implements Exception {
+  /// {@macro cubit_unhandled_error_exception}
+  CubitUnhandledErrorException(this.cubit, this.error, [this.stackTrace]);
+
+  /// The [cubit] in which the unhandled error occurred.
+  final Cubit cubit;
+
+  /// The unhandled [error] object.
+  final Object error;
+
+  /// An optional [stackTrace] which accompanied the error.
+  final StackTrace stackTrace;
+
+  @override
+  String toString() {
+    return 'Unhandled error $error occurred in $cubit.\n'
+        '${stackTrace ?? ''}';
+  }
+}
+
+/// {@template cubit}
+/// A [Cubit] is a subset of [Bloc] which has no notion of events
+/// and relies on methods to [emit] new states.
+///
+/// Every [Cubit] requires an initial state which will be the
+/// state of the [Cubit] before [emit] has been called.
+///
+/// The current state of a [Cubit] can be accessed via the [state] getter.
+///
+/// ```dart
+/// class CounterCubit extends Cubit<int> {
+///   CounterCubit() : super(0);
+///
+///   void increment() => emit(state + 1);
+/// }
+/// ```
+///
+/// {@endtemplate}
+abstract class Cubit<State> extends Stream<State> {
+  /// {@macro cubit}
+  Cubit(this._state);
+
+  /// The current [state].
+  State get state => _state;
+
+  BlocObserver get _observer => Bloc.observer;
+
+  final _controller = StreamController<State>.broadcast();
+
+  State _state;
+
+  bool _emitted = false;
+
+  /// {@template emit}
+  /// Updates the [state] to the provided [state].
+  /// [emit] does nothing if the [Cubit] has been closed or if the
+  /// [state] being emitted is equal to the current [state].
+  ///
+  /// To allow for the possibility of notifying listeners of the initial state,
+  /// emitting a state which is equal to the initial state is allowed as long
+  /// as it is the first thing emitted by the [Cubit].
+  /// {@endtemplate}
+  @protected
```

Comment:
Should include the `@visibleForTesting` annotation just like the override inside `Bloc`.

Reply:
No because then you’d have no way to trigger state changes in your cubits 😄

Reply:
never mind haha I misunderstood your comment and resolved the issue. Thanks for pointing this out! 💯